### PR TITLE
feat(mojaloop#2965): end-end functional tests

### PR DESCRIPTION
**Pull request by sri-miriyala**
feat(mojaloop#2965): end-end functional tests https://github.com/mojaloop/project/issues/2965

### CODE REVIEW

File: modules/api-svc/src/lib/model/InboundTransfersModel.js [hunk 1]
```diff
@@ -501,30 +501,37 @@ class InboundTransfersModel {
             const mojaloopResponse = shared.internalBulkQuotesResponseToMojaloop(response);
 
             // create our ILP packet and condition and tag them on to our internal quote response
+            this._logger.info(JSON.stringify({bulkQuoteRequest}));
```
Comment (mdebarros) [inline modules/api-svc/src/lib/model/InboundTransfersModel.js:504]: I am not sure its a good idea to "stringify" a bulk message like this?

File: modules/api-svc/src/lib/model/lib/shared.js [hunk 2]
```diff
@@ -428,6 +471,61 @@ const internalBulkQuotesResponseToMojaloop = (internal) => {
...
+            // mojaloop payee rejection and put the fsp backend error into an extension
+            // list.
+            if(transfer.errorResponse) {
+                externalTransfer = {
+                    transferId: transfer.transferId,
+                    errorInformation: Errors.MojaloopApiErrorObjectFromCode(Errors.MojaloopApiErrorCodes.PAYEE_REJECTION).errorInformation,
+                    extensionList: {
+                        extension: [
+                            {
+                                key: 'errorResponseStatusCode',
```
Comment (mdebarros) [inline modules/api-svc/src/lib/model/lib/shared.js:499]: I assume this was a mapping that already existed?

File: modules/api-svc/src/lib/model/lib/shared.js [hunk 3]
```diff
@@ -428,6 +471,61 @@ const internalBulkQuotesResponseToMojaloop = (internal) => {
...
+                                value: transfer.errorResponse.statusCode
+                            },
+                            {
+                                key: 'errorResponseMessage',
```
Comment (mdebarros) [inline modules/api-svc/src/lib/model/lib/shared.js:503]: I assume this was a mapping that already existed?

**Review by mdebarros — COMMENTED** [review]
Changes look more or less ok, but I have added a few comments above.

File: modules/api-svc/src/lib/model/lib/shared.js (hunk 2 above)
Comment (kleyow) [reply modules/api-svc/src/lib/model/lib/shared.js:499]: Not that I'm aware of. All the models that send a backend request to a dfsp are either expecting a successful response or some HTTP error. Theres no in between, the backend api makes little consideration for error messages. 

So in bulk where it's a successful response that contains nested quote/transfer errors I have no precedent of a mapping.

I've had lingering thoughts as to why the sdk doesn't just enforce that the backend it talks to use the FSPIOP api. We are putting all the mappings of a backend to the sdk in a flavoured `core-connector` for dfsp's anyway. 

It's hard to design backend api for a non-existent generic fsp.

File: modules/api-svc/src/lib/model/InboundTransfersModel.js (hunk 1 above)
Comment (kleyow) [reply modules/api-svc/src/lib/model/InboundTransfersModel.js:504]: Ah these stringifyies were left overs from debuging. They can be removed.

File: modules/api-svc/src/lib/model/lib/shared.js (hunk 2 above)
Comment (mdebarros) [reply modules/api-svc/src/lib/model/lib/shared.js:499]: We already have something like this here --> https://github.com/mojaloop/central-services-error-handling/blob/master/src/factory.js#L147

Comment (kleyow) [reply modules/api-svc/src/lib/model/lib/shared.js]: Addressed in https://github.com/mojaloop/sdk-scheme-adapter/pull/404

File: modules/api-svc/src/lib/model/lib/shared.js (hunk 3 above)
Comment (kleyow) [reply modules/api-svc/src/lib/model/lib/shared.js]: Added in https://github.com/mojaloop/sdk-scheme-adapter/pull/404

File: modules/api-svc/src/lib/model/InboundTransfersModel.js (hunk 1 above)
Comment (kleyow) [reply modules/api-svc/src/lib/model/InboundTransfersModel.js]: Removed in https://github.com/mojaloop/sdk-scheme-adapter/pull/404

**Review by kleyow — APPROVED** [review]

**Pull request merged**